lib/src/list.dart

```diff
@@ -153,6 +153,18 @@ extension ListX<E> on List<E> {
   void mergeSort({int start = 0, int end, Comparator<E> comparator}) {
     _mergeSort(this, start: start, end: end, compare: comparator);
   }
+
+  /// Swaps the elements in the indices provided.
+  ///
+  /// ```dart
+  /// var list = [1, 2, 3, 4];
+  /// list.swap(0, 2); // [3, 2, 1, 4]
+  /// ```
+  void swap(int i, int j) {
```

Comment:
Please make this function generic and use `T` instead of `int`.

Reply:
like this: `void swap<T extends int>(T i, T j)`? since they are indices afterall and should be `int`s.

Reply:
My dumb brain mad a mistake. `int` is 100% correct.